src/app/shared/services/authentication.service.ts

```diff
@@ -4,18 +4,73 @@ import 'firebase/analytics';
 import 'firebase/auth';
 import 'firebase/firestore';
 import {firebaseConfig} from '../../../firebase-config';
+import {Router} from '@angular/router';
+import {NotifierService} from '../components/notifier/notifier.service';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthenticationService {
 
-  constructor() {
+  provider = new firebase.auth.GoogleAuthProvider();
+
+  constructor(private router: Router,
+              private notifierService: NotifierService) {
     firebase.initializeApp(firebaseConfig);
   }
 
   signUpWithEmailAndPassword(email, password) {
-    firebase.auth().createUserWithEmailAndPassword(email, password).catch((error) => console.log(error)).then(res => console.log(res));
+    firebase.auth().createUserWithEmailAndPassword(email, password).catch((error) => {
+      if (error) {
+        this.notifierService.notify(
+          error.message,
+          2,
+        );
+      }
+    }).then(res => {
+      if (res) {
+        this.redirectToDashboard();
```

Comment:
Auth service should be responsible only for authentication, not redirect. Consider moving redirect logic into signup screen.